esp/useful_scripts/server_setup/new_site.sh

```diff
@@ -421,20 +448,56 @@ fi
 if [[ "$MODE_CRON" || "$MODE_ALL" ]]
 then
-    cat >>$CRON_FILE <<EOF
-* * * * * root $BASEDIR/esp/dbmail_cron.py
+    # Rather than run all dbmail_cron procs every 5 minutes, we choose a random
+    # offset for each.
+    # TODO(benkraft): we should just figure out the right cron syntax to have
+    # cron do this magically, without having to do `sleep` ourselves, and
+    # update both this and existing cron jobs.
+    sleep_sec="$(( (RANDOM % 5) * 60))"
+    if [[ "$sleep_sec" -eq 0 ]] ; then
+        sleep_cmd=""
+    else
+        sleep_cmd="sleep $sleep_sec ; "
+    fi
+    cat >>"$CRON_FILE" <<EOF
+*/5 * * * * root $sleep_cmd$BASEDIR/esp/dbmail_cron.py
 EOF
+
+    # TODO(benkraft): actually echo the added line so there's something to
+    # check.
+    echo "cron has been set up.  Please check the config by looking over the"
+    echo -n "output above, then press enter to continue or Ctrl-C to quit."
+    read THROWAWAY
+fi
+
+if [[ "$MODE_EXIM" || "$MODE_ALL" ]] ; then
+    cat >>"$EXIMDIR/virtual/$ESPHOSTNAME" <<EOF
+*: "|$BASEDIR/esp/mailgates/mailgate.py"
+EOF
+    # The line we want to edit looks like
+    # dc_other_hostnames='foo.learningu.org;bar.learningu.org;baz.learningu.org'
+    # and we want to add another entry to it.
+    sed --in-place=.bak "s/^\\(dc_other_hostnames='.*\\)'$/\\1;$ESPHOSTNAME'/" "$EXIMDIR/update-exim4.conf.conf"
+    update-exim4.conf
+    exim -bt test@$ESPHOSTNAME
+
+    echo "exim4 has been set up.  Please check the config by looking over the"
+    echo -n "output above, then press enter to continue or Ctrl-C to quit."
+    read THROWAWAY
 fi
 
 # Done!
+# Let's let the human do the /etc commit since they may want to separate out
+# other changes, or make sure everything is right, before doing so.
 echo "=== Site setup complete: $ESPHOSTNAME ==="
-IP_ADDRESS=`ifconfig  | grep 'inet addr:'| grep -v '127.0.0.1' | cut -d: -f2 | awk '{ print $1}'`
-echo "Please ensure that DNS is configured to provide A and MX records for:"
-echo "  $ESPHOSTNAME -> $IP_ADDRESS"
-echo "  $SITENAME-orig.learningu.org -> $IP_ADDRESS"
-echo "  $SITENAME-backup.learningu.org -> $IP_ADDRESS"
-echo "You may also want to add some template overrides that establish"
-echo "the initial look and feel of the site."
-echo "Please also configure e-mail by adjusting the exim4 configuration as"
-echo "necessary."
+echo "You may wish to commit your changes to the /etc git repo."
+echo "Please ensure that DNS is configured to direct the correct hostnames"
+echo "to `hostname`.  To do this, log in to the DNS server and edit"
+echo "$DNS_CONF_FILE to increment the serial number in the header, and add"
+echo "the lines:"
+echo "  ${ESPHOSTNAME%%.$DOMAIN} CNAME `hostname`"
+echo "  $SITENAME-orig CNAME `hostname`"
+echo "near the other similar lines.  Then run 'sudo service bind9 restart';"
+echo "after a few minutes you should be able to access the site.  You may"
+echo "wish to also set up a theme and create additional administrators."
 echo
```

Comment:
We could also maybe echo a note that, if the site wants to use Stripe, they should copy it from another site's local_settings.py (obviously we can't commit our Stripe API keys into this script). I don't know whether or not we'll ever have a first-time site using Stripe, so idk if adding that message makes sense or not.